hack/verify-release.sh

```diff
@@ -0,0 +1,576 @@
+)
+
+# quay images
+declare -a container_images=(
+    "${ORG}/baremetal-operator:v${VERSION}"
+    "${ORG}/keepalived:v${VERSION}"
+)
+
+# go mod bump checks - must match up to leading space before v
+declare -A module_groups=(
+    [capi]="
+        sigs.k8s.io/cluster-api
+        sigs.k8s.io/cluster-api/test
+    "
+    [k8s]="
+        k8s.io/api
+        k8s.io/apiextensions-apiserver
+        k8s.io/apimachinery
+        k8s.io/client-go
+        k8s.io/cluster-bootstrap
+        k8s.io/component-base
+    "
+)
+
+# check these modules are using latest patch releases of their releases
+# format: module name=github repo name
+declare -A module_releases=(
+    [sigs.k8s.io/cluster-api]="kubernetes-sigs/cluster-api"
+)
+
+# required tools
+declare -a required_tools=(
+    awk
+    curl
+    git
+    jq
+    sed
+)
+
+# we also require a container runtime, or pre-installed binaries
+if [[ -n "${CONTAINER_RUNTIME}" ]]; then
+    required_tools+=(
+        "${CONTAINER_RUNTIME}"
+    )
+    declare -a GCRANE_CMD=(
+        "${CONTAINER_RUNTIME}" run --rm
+        gcr.io/go-containerregistry/gcrane:latest
+    )
+    declare -a OSVSCANNER_CMD=(
+        "${CONTAINER_RUNTIME}" run --rm -v "${PWD}":/src -w /src
+        ghcr.io/google/osv-scanner:latest
+    )
+else
+    # go install github.com/google/go-containerregistry/cmd/gcrane@latest
+    # go install github.com/google/osv-scanner/cmd/osv-scanner@v1
+    required_tools+=(
+        gcrane
+        osv-scanner
+    )
+    declare -a GCRANE_CMD=(gcrane)
+    declare -a OSVSCANNER_CMD=(osv-scanner)
+fi
+
+
+#
+# temporary files and cleanup trap
+#
+cleanup()
+{
+    rm -rf "${TMP_DIR}"
+}
+
+TMP_DIR="$(mktemp -d "${TMPDIR:-/tmp}/verify-release-XXXXX")"
+RELEASE_JSON="${TMP_DIR}/release.json"
+RELEASES_JSON="${TMP_DIR}/releases.json"
+SCAN_LOG="${TMP_DIR}/scan.log"
+TAG_LOG="${TMP_DIR}/tag.log"
+mkdir -p "${TMP_DIR}"
+trap cleanup EXIT
+
+
+#
+# pre-requisites
+#
+check_tools()
+{
+    echo "Checking required tools ..."
+
+    for tool in "${required_tools[@]}"; do
+        type "${tool}" &>/dev/null || { echo "FATAL: need ${tool} to be installed"; exit 1; }
+    done
+
+    echo -e "Done\n"
+}
+
+detect_remote()
+{
+    # we support origin (default) and upstrea (if cloned with "gh" CLI tool)
+    echo "Detecting remote ..."
+
+    if [[ -z "${REMOTE}" ]]; then
+        REMOTE="$(git remote -v | grep "${PROJECT}.* (fetch)" | awk '{print $1;}')"
+
+        if ! [[ "${REMOTE}" =~ ^(origin|upstream)$ ]]; then
+            echo "WARNING: detected remote '${REMOTE}' is not supported"
+        fi
+    else
+        echo "INFO: Using supplied remote: ${REMOTE}"
+    fi
+
+    echo -e "Done\n"
+}
+
+check_input()
+{
+    echo "Checking input ..."
+
+    # check version is input without leading v, since we have extra annotated
+    # tags in history and it needs manually to be edited out
+    if [[ "${VERSION}" =~ ^v\d+ ]]; then
+        echo "FATAL: given version includes a leading v. Example: 1.5.0"
```

Comment:
I saw a previous example uses `Example: 0.4.0` shall we use the same here to relate to BMO release ?

Reply:
Good catch, fixed.